File: 2022/04/js/1.js
const fs = require("fs");
let data = fs.readFileSync("i.txt").toString().split(/\r?\n/);
console.log(data.reduce((acc, l) => {
    [a, b, c, d] = l.split(/,|-/).map(c => parseInt(c, 10));
    if ((a <= c && b >= d) || (c <= a && d >= b)) return acc + 1;
    return acc;
}, 0));

File: 2022/04/py/1.py
import re
data = open("i.txt").read().split("\n")

n = 0
for l in data:
    m = re.search(r"(\d*)-(\d*),(\d*)-(\d*)", l)
    a, b, c, d = m.groups()
    if len(set(range(int(a),int(b)+1)).intersection(range(int(c),int(d)+1)))!=0:
        n+=1
print(n)